src/qmapshack/gis/trk/CTrackData.cpp

```diff
@@ -140,6 +140,39 @@ void CTrackData::getPolyline(SGisLine& l) const
     }
 }
 
+void CTrackData::getPolylineRange(SGisLine& l, qint32 rangeStart, qint32 rangeEnd, bool getSubPixel) const
+{
+    l.clear();
+    for (const trkseg_t &cur_seg : segs)
+    {
+        for (const trkpt_t &pt : cur_seg.pts)
```

Comment:
Naming convention. no snake_case. camelCase.  In that case `seg` is totally fine.